freeipa/src/main/java/com/sequenceiq/freeipa/flow/stack/termination/action/TerminationService.java

```diff
@@ -54,22 +54,22 @@
     private KerberosMgmtV1Service kerberosMgmtV1Service;
 
     public void finalizeTermination(Long stackId) {
-        Stack stack = stackService.getByIdWithListsInTransaction(stackId);
-        Date now = new Date();
-        String terminatedName = stack.getName() + DELIMITER + now.getTime();
+        long currentTimeMillis = clock.getCurrentTimeMillis();
         try {
             transactionService.required(() -> {
+                Stack stack = stackService.getByIdWithListsInTransaction(stackId);
+                String terminatedName = stack.getName() + DELIMITER + currentTimeMillis;
                 stack.setName(terminatedName);
-                stack.setTerminated(clock.getCurrentTimeMillis());
+                stack.setTerminated(currentTimeMillis);
                 terminateInstanceGroups(stack);
                 terminateMetaDataInstances(stack);
                 cleanupVault(stack);
+                stackUpdater.updateStackStatus(stack, DetailedStackStatus.DELETE_COMPLETED, "Stack was terminated successfully.");
                 stackService.save(stack);
-                stackUpdater.updateStackStatus(stackId, DetailedStackStatus.DELETE_COMPLETED, "Stack was terminated successfully.");
                 return null;
             });
-        } catch (TransactionService.TransactionExecutionException ex) {
-            LOGGER.info("Failed to terminate cluster infrastructure. Stack id {}", stack.getId());
+        } catch (TransactionExecutionException ex) {
+            LOGGER.info("Failed to terminate cluster infrastructure.");
```

Comment:
nit: do we refer to freeipa stacks as "cluster" elsewhere? I think it would be clearer to refer to this as "freeipa infrastructure"

stackId is still available as a method parameter if you want to continue logging it.

Reply:
it's a copy-paste error. yeah I could have added it, but hopefully the logcontext already contains that information, so I didn't think much about deleting it